pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/OverrideBothEqualsAndHashCodeOnComparableRule.java

```diff
@@ -30,24 +30,29 @@ protected boolean skipType(ASTTypeDeclaration node) {
         return !TypeTestUtil.isA(Comparable.class, node) || TypeTestUtil.isA(Enum.class, node);
     }
 
-    private static boolean isCompareToMethod(ASTMethodDeclaration method) {
-        return "compareTo".equals(method.getName())
-                && method.getArity() == 1
-                && method.getResultTypeNode().getTypeMirror().isPrimitive(JPrimitiveType.PrimitiveTypeKind.INT)
-                && !method.isStatic();
+    private static boolean hasBrokenEqualsMethod(ASTTypeDeclaration node) {
```

Comment:
What is a "broken equals method"? I don't see that we have a test case for this...